### ISSUE TITLE
Results with the normalized Laplacian

### ISSUE THREAD

**Issue by jongharyu** [shaham-lab/SpectralNet#6]
I have a question regarding the implementation and performance of the normalized Laplacian version.

First of all, in the paper, the normalized Laplacian version objective is written as Eq. (5), which is:
<img width="752" alt="image" src="https://github.com/shaham-lab/SpectralNet/assets/18149073/3f2c356f-dc6e-4dc4-b674-fffc29cac9ee">
To my understanding, however, the normalization should be $\frac{y_i}{\sqrt{d_i}}$ in place of $\frac{y_i}{d_i}$, as otherwise the orthogonality structure would be different from decomposing the normalized Laplacian. And I can see that the current code is implemented based on this expression. Is there a reasoning behind this choice?

Secondly, I wonder if there is any rule of thumb in the choice of which Laplacian to use in practice. 
In other words, are there circumstances where the normalized Laplacian gives better results than the unnormalized one?
If I am not misunderstood, it seems that the paper only reports the unnormalized version.
I did some experiments with the two moons dataset and found that the normalized version (both with the original normalization or with the square root normalization) does not work well.
Any advice would be greatly appreciated!

**Comment by AmitaiYacobi** [comment]
Hi again!
First, you are completely correct and the normalization should be $\frac{y_i}{\sqrt{di}}$ as you mentioned.  I will fix the implementation as well.
Second, I am not familiar with such a rule of thumb, but maybe you would like to use the normalized version when dealing with graphs or affinity matrices that have varying degrees of connectivity between nodes, or when the size of the clusters is significantly different which is completely not the case in the two moons dataset.

You can find more relevant information in the following well-known tutorial:
[Luxburg07_tutorial_spectral_clustering](https://people.csail.mit.edu/dsontag/courses/ml14/notes/Luxburg07_tutorial_spectral_clustering.pdf)

Hope that will help!
Let me know if you have any other questions!

**Comment by jongharyu** [comment]
I appreciate your reply! Thank you for confirming that the normalization was by mistake. 
For the different types of Laplacian, I get the explanation. Thanks again and I will get back to you if I have another question!
